### ISSUE TITLE
[v2] The host and routingKey fields are not inheriting from name

### ISSUE THREAD

**Issue by nluaces** [skupperproject/skupper#1609]
**Is your feature request related to a problem? Please describe.**
The host and routingKey fields are not inheriting from name

```
$ skupper listener create backend 8080 -o yaml
apiVersion: skupper.io/v1alpha1
kind: Listener
metadata:
  creationTimestamp: null
  name: backend
  namespace: hw
spec:
  host: ""          # <<<
  port: 8080
  routingKey: ""    # <<<
  type: tcp
status: {} 
```
**Describe the solution you'd like**
The host and routingKey fields default to the value of name, supplied in the example as "backend".

**Comment by lynnemorrison** [comment]
I've updated the listener command to user the listener's name as the host and routingkey if either is not defined.
	
@ssorj With the connector, I have questions from what is defined in https://skupperproject.github.io/refdog/commands/connector/create.html
1. workload, selector or host must be specified. 
2. if host is not defined default is "connector-name"
3. if selector is not defined default is "app=connector-name"
4. workload is an alternative to setting the --selector or --host options.

So if host/selector/workload is not set what is the preference:
1. just set host = "connector-name"
2. just set selector = "app=connector-name"
3. set both 1 and 2?

What happens if user sets host and selector and/or workload?

**Comment by ssorj** [comment]
> I've updated the listener command to use the listener's name as the host and routingkey if either is not defined.
> 
> @ssorj With the connector, I have questions from what is defined in https://skupperproject.github.io/refdog/commands/connector/create.html
> 
> 1. workload, selector or host must be specified.
> 2. if host is not defined default is "connector-name"
> 3. if selector is not defined default is "app=connector-name"
> 4. workload is an alternative to setting the --selector or --host options.
> 
> So if host/selector/workload are not set what is the preference:
> 
> 1. just set host = "connector-name"
> 2. just set selector = "app=connector-name"
> 3. set both 1 and 2?  Is this allowed?

Item 3 is not allowed.  Item 2 is what we want.  (We might need a different default for non-Kube platforms, but that's another issue.)

> What happens if user sets host and selector and/or workload? I believe only one option is allowed.

It should be an error to specify more than one of the three.

**Comment by lynnemorrison** [comment]
code complete waiting on this PR to be merged
https://github.com/skupperproject/skupper/pull/1563
